File: loadercomplex-forge-18/src/main/java/com/enderzombi102/loadercomplex/forge18/impl/item/ForgeItemStack.java
package com.enderzombi102.loadercomplex.forge18.impl.item;

import com.enderzombi102.loadercomplex.api.item.ItemStack;
import net.minecraft.text.LiteralText;

public class ForgeItemStack implements ItemStack {

	private final net.minecraft.item.ItemStack stack;

	public ForgeItemStack(net.minecraft.item.ItemStack stack) {
		this.stack = stack;
	}

	public net.minecraft.item.ItemStack getStack() {
		return stack;
	}

	@Override
	public String getDisplayName() {
		return this.stack.getName().getString();
	}

	@Override
	public void setDisplayName(String newName) {
		this.stack.setCustomName( new LiteralText( newName ) );
	}

	@Override
	public Object getType() {
		return this.stack.getItem();
	}

	@Override
	public int getAmount() {
		return this.stack.getCount();
	}

	@Override
	public void setAmount(int amount) {
		this.stack.setCount(amount);
	}

	@Override
	public void setDamage(int damage) {
		this.stack.setDamage(damage);
	}

	@Override
	public int getDamage() {
		return this.stack.getDamage();
	}

	@Override
	public int getMaxDurability() {
		return this.stack.getMaxDamage();
	}

	@Override
	public int getMaxStackSize() {
		return this.stack.getMaxCount();
	}
}

File: loadercomplex-forge/src/ver182/kotlin/com/enderzombi102/loadercomplex/forge182/impl/item/ForgeItemStack.kt
package com.enderzombi102.loadercomplex.forge182.impl.item

import net.minecraft.item.ItemStack
import net.minecraft.text.LiteralText

class ForgeItemStack(private val stack: ItemStack) : com.enderzombi102.loadercomplex.api.minecraft.item.ItemStack {
	override fun getStack(): Any {
		return stack
	}

	override fun getDisplayName(): String? {
		return stack.name.string
	}

	override fun setDisplayName(newName: String) {
		stack.setCustomName(LiteralText(newName))
	}

	override fun getType(): Any {
		return stack.item
	}

	override fun getAmount(): Int {
		return stack.count
	}

	override fun setAmount(amount: Int) {
		stack.count = amount
	}

	override fun setDamage(damage: Int) {
		stack.damage = damage
	}

	override fun getDamage(): Int {
		return stack.damage
	}

	override fun getMaxDurability(): Int {
		return stack.maxDamage
	}

	override fun getMaxStackSize(): Int {
		return stack.maxCount
	}
}
